### DISCUSSION TITLE
Install mws without graphql

### DISCUSSION THREAD

**Discussion by Flui** [mswjs/msw#1167] [Q&A]
### Scope

Adds a new behavior

### Compatibility

- [X] This is a breaking change

### Feature description

**Is your feature request related to a problem? Please describe.**

Currently I am working on an Project with react-scripts@4.0.3 and typescript@3.8.3. After installing latest version of msw I get the following typescript error:
`"Package 'graphql' support only TS versions that are >=4.1.0"`
As I am consuming a rest api via fetch, there is no need for me here to even install graphql.

**Describe the solution you'd like**

So I would like to install msw with the minimal required dependencies to use it for rest apis.

**Describe alternatives you've considered**

If somebody has a workaround without updating typescript I would appreciate that as well.

**Comment by kettanaito** [comment]
Hey, @Flui. 

While I understand your concern, MSW is a devtool, and its dependencies are bound to the goals it accomplishes. I understand you may not use GraphQL mocking in your project. Yet the price you pay for installing an extra package is rather small. 

In a similar fashion, we're installing packages to manage cookies, which you may not mock. There's no agreed way to modularize libraries in Node.js, so you will end up installing more than you may potentially use _and it's fine_. 

If you get a warning from a tool you're not using, you can just ignore that warning. The one you're getting is interesting though, because it suggests you're using TypeScript < 4.1.0. MSW requires TypeScript 4.3 to run, so you have to update TypeScript if you want to use MSW without issues. 

> We are working on communicating the required TypeScript version more clearly in #985.

**Comment by IlyaShorin** [comment]
have same issue, 

`npm ls typescript` shows me 

```bash
─┬ @typescript-eslint/eslint-plugin@7.18.0
│ └─┬ ts-api-utils@1.4.3
│   └── typescript@5.5.4 deduped
├─┬ msw@2.7.0
│ └── typescript@5.5.4 deduped
├─┬ stylelint@16.14.0
│ └─┬ cosmiconfig@9.0.0
│   └── typescript@5.5.4 deduped
├── typescript@5.5.4
└─┬ vite-plugin-checker@0.8.0
  └── typescript@5.5.4 deduped
  ```
  
  local build with vite is successfull, but in CI i get the error same as above

**Reply by IlyaShorin** [reply]
oops, my problem is not exactly the same. 

```
 x Unexpected token
   ,-[node_modules/graphql/NotSupportedTSVersion.d.ts:1:63]
 1 | "Package 'graphql' support only TS versions that are >=4.1.0".
 ```
 
 probably something with text, not version itself...

**Reply by IlyaShorin** [reply]
would appreciate if someone could help

**Reply by IlyaShorin** [reply]
if anyone will have same problem, the thing that has helped me it's adding in my 
`oxlintrc.json`
in ignorePattern not only 
"node_modules/**",

but also 
"node_modules",

**Reply by kettanaito** [reply]
This is a good time to mention that the minimal TypeScript version to use with MSW is 4.8, which is also the oldest TS version that's not EOL. Updating may help.
